walk/config.go

```diff
@@ -49,15 +54,22 @@ func getWalkConfig(c *config.Config) *walkConfig {
 	return c.Exts[walkName].(*walkConfig)
 }
 
-func (wc *walkConfig) isExcluded(rel, base string) bool {
-	if base == ".git" {
-		return true
-	}
-	return matchAnyGlob(wc.excludes, path.Join(rel, base))
+func (wc *walkConfig) isExcluded(p string) bool {
+	return matchAnyGlob(wc.excludes, p)
+}
+
+func (wc *walkConfig) shouldFollow(p string) bool {
+	return matchAnyGlob(wc.follow, p)
 }
 
-func (wc *walkConfig) shouldFollow(rel, base string) bool {
-	return matchAnyGlob(wc.follow, path.Join(rel, base))
+func (wc *walkConfig) isGitIgnored(p []string, isDir bool) bool {
+	if wc.gitignoreEnabled && wc.gitignore != nil {
```

Comment:
do you actually need the `wc.gitignore != nil` check? It looks like a nil receiver will be handled correctly? https://github.com/go-git/go-git/blob/cef892e5701bb6ae7c740e088ad4c19894f2fd60/plumbing/format/gitignore/matcher.go#L18-L30

Reply:
It looks like you're correct, so we can make this method 1 line 👍